### PR TITLE
chore: oidc v2 proto definitions

### PR DESCRIPTION
**Pull request by muhlemmer**
This PR adds proto definitions for the API as defined in #5468.

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] ~~Critical parts are tested automatically~~
- [ ] ~~Where possible E2E tests are implemented~~
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] ~~Functionality of the acceptance criteria is checked manually on the dev system.~~

### CODE REVIEW

File: proto/zitadel/oidc/v2alpha/authorization.proto [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+option go_package = "github.com/zitadel/zitadel/pkg/grpc/oidc/v2alpha;oidc";
+
+message AuthRequest{
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+    external_docs: {
+      url: "https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest";
+      description: "Find out more about OIDC Auth Request parameters";
+    }
+  };
```
Comment (muhlemmer) [inline proto/zitadel/oidc/v2alpha/authorization.proto:17]: This top-level schema documentation does not get rendered somehow:
https://docs-f7o3cd32k-zitadel.vercel.app/docs/apis/resources/oidc_service/oidc-service-get-auth-request

But it is valid syntax as per this example:

https://github.com/grpc-ecosystem/grpc-gateway/blob/06fae32b118ceb06e71790117a41c323f70de03e/examples/internal/proto/examplepb/a_bit_of_everything.proto#L235-L238

Is it a bug or should this be handeled differently?

Comment (fforootd) [reply proto/zitadel/oidc/v2alpha/authorization.proto:17]: Hm on a first glance it looks correct.

Let me check if the generated output contains the url.

File: proto/zitadel/oidc/v2alpha/authorization.proto [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Time when the auth request was created";
+    }
+  ];
+
+  string application_id = 3 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "OIDC client ID of the application that created the auth request";
+    }
+  ];
```
Comment (livio-a) [inline proto/zitadel/oidc/v2alpha/authorization.proto:35]: the application_id probably originates from the current domain.AuthRequest which can be either the OIDC client_id or SAML entity_id
as this service is OIDC only i think we should go with client_id

File: proto/zitadel/oidc/v2alpha/authorization.proto [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Requested scopes by the application, which the user must consent to.";
+    }
+  ];
+  
+  string callback_uri = 5 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Base URI that points back to the application";
+    }
+  ];
```
Comment (livio-a) [inline proto/zitadel/oidc/v2alpha/authorization.proto:47]: based on the comment above, should we go with redirect_uri?

File: proto/zitadel/oidc/v2alpha/authorization.proto [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  repeated Prompt prompt = 6 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Prompts that must be displayed to the user";
+    }
+  ];
+
+  repeated string ui_locales = 7 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "End-User's preferred languages and scripts for the user interface, represented as a space-separated list of BCP47 [RFC5646] language tag values, ordered by preference. For instance, the value \"fr-CA fr en\" represents a preference for French as spoken in Canada, then French (without a region designation), followed by English (without a region designation). An error SHOULD NOT result if some or all of the requested locales are not supported.";
+    }
```
Comment (livio-a) [inline proto/zitadel/oidc/v2alpha/authorization.proto:58]:
```suggestion
      description: "End-User's preferred languages and scripts for the user interface, represented as a list of BCP47 [RFC5646] language tag values, ordered by preference. For instance, the value [fr-CA, fr, en] represents a preference for French as spoken in Canada, then French (without a region designation), followed by English (without a region designation). An error SHOULD NOT result if some or all of the requested locales are not supported.";
    }
```

Comment (livio-a) [reply proto/zitadel/oidc/v2alpha/authorization.proto:58]: as we already parse the string into a list / array

File: proto/zitadel/oidc/v2alpha/oidc_service.proto [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+message GetAuthRequestRequest {
+  string auth_request_id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      description: "ID of the Auth Request, as obtained from the redirect URL.";
+      example: "\"d654e6ba-70a3-48ef-a95d-37c8d8a7901a\"";
```
Comment (livio-a) [inline proto/zitadel/oidc/v2alpha/oidc_service.proto:131]:
```suggestion
      example: "\"163840776835432705\"";
```

Comment (livio-a) [reply proto/zitadel/oidc/v2alpha/oidc_service.proto:131]: Fabi noticed that we should use sonyflake style ids in the examples

File: proto/zitadel/oidc/v2alpha/oidc_service.proto [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+message CreateCallbackRequest {
+  string auth_request_id = 1 [
+    (validate.rules).string = {min_len: 1, max_len: 200},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      min_length: 1;
+      max_length: 200;
+      description: "ID of the Auth Request, as obtained from the redirect URL.";
+      example: "\"d654e6ba-70a3-48ef-a95d-37c8d8a7901a\"";
```
Comment (livio-a) [inline proto/zitadel/oidc/v2alpha/oidc_service.proto:147]:
```suggestion
      example: "\"163840776835432705\"";
```

**Review by livio-a — CHANGES_REQUESTED** [review]
some minor changes... i'm also always unsure about `scope` vs `scopes` and `prompt` vs `prompts`

File: proto/zitadel/oidc/v2alpha/authorization.proto (hunk 2 above)
Comment (muhlemmer) [reply proto/zitadel/oidc/v2alpha/authorization.proto:35]:
```suggestion
  string client_id = 3 [
    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
      description: "OIDC client ID of the application that created the auth request";
    }
  ];
```

File: proto/zitadel/oidc/v2alpha/authorization.proto (hunk 3 above)
Comment (muhlemmer) [reply proto/zitadel/oidc/v2alpha/authorization.proto:47]:
```suggestion
  string redirect_uri = 5 [
    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
      description: "Base URI that points back to the application";
    }
  ];
```

**Review by livio-a — APPROVED** [review]

**Pull request merged**